Feature 6 trading

colewiltse opened the pull request:
Here is my implementation for the Trade use case.

JLenander commented on src/main/java/TurnUseCases/TradeUseCase/Trade.java:
```diff
@@ -0,0 +1,130 @@
+ */
+
+public class Trade implements  TradeInputBoundary{
+    TemporaryTurnActionPresenter presenter = new TemporaryTurnActionPresenter();
```
The `presenter` here should be a object of type `TradeOutputBoundary` that is passed in to the constructor for `Trade`. This way you don't need the `TemporaryTurnActionPresenter` class.

colewiltse replied:
Just added a constructor that does this.

JLenander commented on src/test/java/TurnUseCases/TradeUseCase/TradeOfferTest.java:
```diff
@@ -0,0 +1,156 @@
+public class TradeOfferTest {
+    Player player1 = new Player("player1", "player1", 1500);
+
+    ColorPropertyTile fakeStreet = new ColorPropertyTile("red", "fakeStreet",
```
`ColorPropertyTile has been updated to use an array for rent instead of different parameters as of e40f2aae15067f77e9d9bc59dfa160ec495e37f5

JLenander reviewed (changes requested):
`ColorPropertyTile` has been updated to use an array for rent instead of different parameters as of e40f2aae15067f77e9d9bc59dfa160ec495e37f5 (tests are using old constructor)
Get out of jail free cards are tradable and so should be represented in the options.

colewiltse commented:
Added the option to trade with get out of jail free cards and updated my tests to work with both this new feature as well as the updated constructor for colour property.

JLenander commented:
Looks like your input boundary (and other class) methods don't use camelCase. I'm also a little confused as to what the input boundary methods are doing (They say they provide the presenter with some information but also return items to the Controller).

alirezazim03 commented on src/main/java/TurnUseCases/TradeUseCase/TradeOffer.java:
```diff
@@ -0,0 +1,153 @@
+     *
+     * @return whether the player offering the jail card has one.
+     */
+    public boolean CheckJailCard(){
```
the player class does have a method that will return true if the player owns a get out of jail free card

alirezazim03 replied:
its called hasGetOutofJailFreeCard

colewiltse commented:
Just fixed the javadocs as well as renamed my methods to use camelCase.
